# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/tests/gtest/TestThreadMetrics.cpp
# Commit: 81d537df2dc1
# Full Hash: 81d537df2dc1f94936e0fbc5542c34803bb4064f
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 09:35:42
# Regressor Bug: 1580565
# File Overlap Count: 2
# Description:
#   Bug 1580565 - Part 6: Add a unique ID to each BrowsingContextGroup, r=kmag
#   
#   This allows us to explicitly specify BrowsingContextGroups when synchronizing
#   them. A major advantage of this is that it means we can handle an attempt to
#   create a BrowsingContext with a parent which the content process is unaware of,
# ==============================================================================

diff -r b182e872c9d4 -r 81d537df2dc1 xpcom/tests/gtest/TestThreadMetrics.cpp
--- a/xpcom/tests/gtest/TestThreadMetrics.cpp	Wed Apr 22 01:38:02 2020 +0000
+++ b/xpcom/tests/gtest/TestThreadMetrics.cpp	Wed Apr 22 01:38:05 2020 +0000
@@ -118,7 +118,8 @@
  protected:
   virtual void SetUp() {
     // building the DocGroup structure
-    RefPtr<dom::BrowsingContextGroup> group = new dom::BrowsingContextGroup();
+    RefPtr<dom::BrowsingContextGroup> group =
+        dom::BrowsingContextGroup::Create();
     mDocGroup = group->AddDocument(NS_LITERAL_CSTRING("key"), nullptr);
     mDocGroup2 = group->AddDocument(NS_LITERAL_CSTRING("key2"), nullptr);
     mCounter = mDocGroup->GetPerformanceCounter();
